internal/weldr/api.go

```diff
@@ -2443,6 +2456,41 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 		return
 	}
 
+	if archName != api.hostArch {
+		log.Println("Cross-architecture build, checking if remote worker socket is active")
+		sdconn, err := dbus.NewWithContext(context.Background())
+		if err != nil {
+			log.Println("Error looking at systemd:", err)
+		}
+		units, err := sdconn.ListUnitsByNamesContext(context.Background(), []string{"osbuild-remote-worker.socket"})
+		if err != nil {
+			log.Println("Error listing units:", err)
+		}
+		if len(units) > 1 {
+			log.Println("Found multiple units with the name \"osbuild-remote-worker.socket\"")
+		}
+		if len(units) == 0 {
+			log.Println("Cross-architecture build requested while \"osbuild-remote-worker.socket\" wasn't found, no remote workers can connect")
```

Comment:
What action would normally fix this? Do they need to manually run `systemctl start osbuild-remote-worker.socket`?

Reply:
Well the idea is that if the remote socket is inactive, there are no remote workers up, meaning cross-arch builds won't be run. It's a proxy for not having a clear list of which workers have called in. We could also maintain a list of workers which have called into the worker api of course.

Reply:
So to fix this, attach a worker, right?

Nitpicks incoming:

Workers could all be on the local socket ... Either by piping it over SSH or by having a local worker on emulated hardware (qemu-user or in a VM).

Not that that's something we should go out of our way to support, but maybe also not something to break on purpose?

Also, we currently allow workers to be attached after jobs have been queued, so even if we have remote workers they could just be "late".

How about: track the workers and add a new endpoint to query the state. Then the cli/UI could show an informational message "So... I did queue that job as you asked, but you better get a worker to connect or you won't have a good time..." (maybe run the phrasing of the message by someone else).

Reply:
So tracking the workers is also a best-effort kinda thing right? it would look at the workers which called in within the last x hours and if so let you queue it. So if a worker is 'late' we'd have the same problem. It is a nicer solution though I agree.

Reply:
I was thinking of it more along the lines of me seeing this in the logs and not having any idea how to make sure it has the workers I need, and not knowing how to start them, etc.

Reply:
Sanne: don't we know all the pending workers at any given time?

Brain: very good point, not very actionable as currently written.

Reply:
Ah right we do know the workers which have called in, but then I wonder how to do his in a distributed setup, like psql really doesn't feel like the place to track this.

Reply:
That said the workers don't have a unique identifier, so that means the list is just 'calls on the request job endpoint'.

Reply:
Looking at it further,  tracking the workers would require breaking the worker api to uniquely identify each worker, unless we simply want to keep track of `$arch job was requested x time ago`, one entry per arch. So I don't even know if we should have that in the db.